src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisConfigForm.tsx

```diff
@@ -0,0 +1,354 @@
+    (templateName?: string, templateType?: TemplateType) => {
+      setTemplateName(templateName);
+      setTemplateType(templateType);
+    },
+    [setTemplateName, setTemplateType]
+  );
+
+  const getEventString = React.useCallback(() => {
+    var str = '';
+    if (!!templateName) {
+      str += `template=${templateName}`;
+    }
+    if (!!templateType) {
+      str += `,type=${templateType}`;
+    }
+    return str;
+  }, [templateName, templateType]);
+
+  const handleSubmit = React.useCallback(() => {
+    setIsLoading(true);
+    const options: RecordingOptions = {
+      toDisk: true,
+      maxAge: maxAge * maxAgeUnits,
+      maxSize: maxSize * maxSizeUnits,
+    };
+    const recordingAttributes: RecordingAttributes = {
+      name: automatedAnalysisRecordingName,
+      events: getEventString(),
+      duration: undefined,
+      archiveOnStop: false,
+      options: options,
+      metadata: {
+        labels: {
+          origin: automatedAnalysisRecordingName,
+        },
+      },
+    };
+    addSubscription(
+      context.api
+        .createRecording(recordingAttributes)
+        .pipe(first())
+        .subscribe({
+          next: (resp) => {
+            if (resp.ok && props.onCreate) {
+              props.onCreate();
+            }
+            setIsLoading(false);
+          },
+          error: (err) => {
+            setIsLoading(false);
+          },
+        })
+    );
+  }, [
+    addSubscription,
+    context.api,
+    setIsLoading,
+    getEventString,
+    props.onCreate,
+    maxAge,
+    maxAgeUnits,
+    maxSize,
+    maxSizeUnits,
+  ]);
+
+  const handleSaveConfig = React.useCallback(() => {
+    const options: AutomatedAnalysisRecordingConfig = {
+      templates: getEventString(),
+      maxAge: maxAge * maxAgeUnits,
+      maxSize: maxSize * maxSizeUnits,
+    };
+    context.settings.setAutomatedAnalysisRecordingConfig(options);
+    setIsSaveLoading(true);
+    const timer = setTimeout(() => {
+      setShowHelperMessage(true);
+      if (props.onSave) {
+        props.onSave();
+      }
+      setIsSaveLoading(false);
+    }, 1000);
+    return () => clearInterval(timer);
+  }, [
+    getEventString,
+    setIsSaveLoading,
+    setShowHelperMessage,
+    props.onSave,
+    maxAge,
+    maxAgeUnits,
+    maxSize,
+    maxSizeUnits,
+    context.settings,
+    context.settings.setAutomatedAnalysisRecordingConfig,
+  ]);
+
+  return (
+    <Form isHorizontal={props.isSettingsForm}>
+      <FormSection title={props.isSettingsForm ? '' : 'Profiling Recording Configuration'}>
+        <FormGroup
+          label="Template"
+          isRequired
+          fieldId="recording-template"
+          validated={!templateName ? ValidatedOptions.default : ValidatedOptions.success}
+          helperText={'The Event Template to be applied in this recording'}
+          helperTextInvalid="A Template must be selected"
+        >
+          <SelectTemplateSelectorForm
```

Comment:
Just a heads up that #684 changed the props to add a required `selected` field to `SelectTemplateSelectorForm`.

Reply:
Hey @maxcao13, I think you can start rebasing and add the selected props?